lib/pkgcloud/openstack/network/client/securityGroupRules.js

```diff
@@ -0,0 +1,156 @@
+ * for Openstack Networking
+ *
+ * (C) 2015 Rackspace
+ *      Shaunak Kashyap
+ * MIT LICENSE
+ */
+
+var urlJoin = require('url-join');
+
+var securityGroupRulesResourcePath = '/security-group-rules';
+
+// Declaring variables for helper functions defined later
+var _convertSecurityGroupRuleToWireFormat;
+
+/**
+ * client.getSecurityGroupRules
+ *
+ * @description get the list of security group rules for an account
+ *
+ * @param {object|Function}   options
+ * @param {Function}          callback
+ */
+exports.getSecurityGroupRules = function (options, callback) {
+  var self = this;
+
+  if (typeof options === 'function') {
+    callback = options;
+    options = {};
+  }
+
+  var getSecurityGroupRuleOpts = {
+    path: securityGroupRulesResourcePath
+  };
+
+  this._request(getSecurityGroupRuleOpts, function (err, body) {
+    if (err) {
+      return callback(err);
+    }
+    else if (!body || !body.security_group_rules || !(body.security_group_rules instanceof Array)) {
+      return new Error('Malformed API Response');
+    }
+
+    return callback(null, body.security_group_rules.map(function (securityGroupRule) {
+      return new self.models.SecurityGroupRule(self, securityGroupRule);
+    }));
+  });
+};
+
+/**
+ * client.getSecurityGroupRule
+ *
+ * @description get the details for a specific securityGroupRule
+ *
+ * @param {String|object}     securityGroupRule     the securityGroupRule or securityGroupRuleId
+ * @param callback
+ */
+exports.getSecurityGroupRule = function (securityGroupRule, callback) {
+  var securityGroupRuleId = securityGroupRule instanceof this.models.SecurityGroupRule ? securityGroupRule.id : securityGroupRule,
+    self = this;
+  self.emit('log::trace', 'Getting details for security group rule', securityGroupRuleId);
+  this._request({
+    path: urlJoin(securityGroupRulesResourcePath, securityGroupRuleId),
+    method: 'GET'
+  }, function (err, body) {
+    if (err) {
+      return callback(err);
+    }
+
+    if (!body ||!body.security_group_rule) {
+      return new Error('Malformed API Response');
+    }
+
+    callback(null, new self.models.SecurityGroupRule(self, body.security_group_rule));
+  });
+};
+
+/**
+ * client.createSecurityGroupRule
+ *
+ * @description create a new securityGroupRule
+ *
+ * @param object      securityGroupRule
+ * @param string      securityGroupRule.securityGroupId The security group ID to associate with this security group rule.
+ * @param string      securityGroupRule.direction       The direction ("ingress" or "egress") in which the security group rule is applied.
+ * @param {string}    securityGroupRule.ethertype       "IPv4" or "IPv6".
+ * @param {int}       securityGroupRule.portRangeMin    The minimum port number in the range that is matched by the security group rule.
+ * @param {int}       securityGroupRule.portRangeMax    The maximum port number in the range that is matched by the security group rule.
```

Comment:
Question: often in my jsdoc i use `{number}` instead of {int} as it's what's returned for `typeof` checks. Is there an advantage to using `int` here?